### PR TITLE
#5697 Feat: added promise with limit logic

### LINKED ISSUE #5697: "Too Many Requests: 429 Error" when executing guessContactsFromSentEmails
Sometimes when composing message, after I start entering recipient email appears error popup `Error searching contacts` and there are such errors in console:

```
catch.js:45 Too Many Requests: 429 when GET-ing https://www.googleapis.com/gmail/v1/users/me/messages/188b462c4189b5a1 (no body):  -> Resource has been exhausted (e.g. check quota).


provided ajax call stack:
    at Google.gmailCall (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/google.js:40:67)
    at async Gmail.msgGet (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:161:24)
    at async Promise.all (index 46)
    at async Gmail.msgsGet (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:169:24)
    at async Gmail.extractHeadersFromMsgs (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:529:35)
    at async Gmail.fetchMsgsHeadersBasedOnQuery (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:396:24)
    at async Gmail.apiGmailLoopThroughEmailsToCompileContacts (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:497:37)
    at async Gmail.guessContactsFromSentEmails (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/common/api/email-provider/gmail/gmail.js:375:17)

url: chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/chrome/elements/compose.htm?frameId=frame_rWjBxgpsDx&acctEmail=[SCRUBBED]&parentTabId=cs.BcSjCrXnsT

    at <async> ajax (/extension/js/common/api/shared/api.ts:98:4)
    at <async> performAjaxRequest (/extension/js/common/api/authentication/google/google-oauth.ts:128:31)
    at <async> gmailCall (/extension/js/common/api/email-provider/gmail/google.ts:19:28)
    at <async> msgGet (/extension/js/common/api/email-provider/gmail/gmail.ts:128:18)
    at <async> msgsGet (/extension/js/common/api/email-provider/gmail/gmail.ts:137:19)
    at <async> extractHeadersFromMsgs (/extension/js/common/api/email-provider/gmail/gmail.ts:454:35)
    at <async> fetchMsgsHeadersBasedOnQuery (/extension/js/common/api/email-provider/gmail/gmail.ts:333:40)
    at <async> apiGmailLoopThroughEmailsToCompileContacts (/extension/js/common/api/email-provider/gmail/gmail.ts:416:55)
    at <async> guessContactsFromSentEmails (/extension/js/common/api/email-provider/gmail/gmail.ts:282:39)
```

There are also several failed requests trying to fetch messages metadata (https://www.googleapis.com/gmail/v1/users/me/messages/MESSAGE_ID?format=metadata
Screenshot 2024-04-30 at 21 18 51

Currently noticed such issue only in https://github.com/FlowCrypt/flowcrypt-browser/pull/5651, but maybe it exists in `master` too, can't find exact steps to reproduce it, occurs randomly when composing new message.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR added promise with limit logic to prevent 429 errors

close #5697 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/api/email-provider/gmail/gmail.ts [hunk 1]
```diff
@@ -136,7 +136,10 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
   };
 
   public msgsGet = async (msgIds: string[], format: GmailResponseFormat): Promise<GmailRes.GmailMsg[]> => {
-    return await Promise.all(msgIds.map(id => this.msgGet(id, format)));
+    return await promiseAllWithLimit(
+      10,
```
Comment (sosnovsky) [inline extension/js/common/api/email-provider/gmail/gmail.ts:140]: According to Gmail usage limits it's possible to have 50 concurrent `msg.get` requests - https://developers.google.com/gmail/api/reference/quota
What about increasing current 10 requests limit to 30?

**Review by sosnovsky — COMMENTED** [review]
Works well 👍

File: extension/js/common/api/email-provider/gmail/gmail.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/api/email-provider/gmail/gmail.ts:140]: Sounds good!

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**